wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/ChePlugin.java

```diff
@@ -11,36 +11,126 @@
  */
 package org.eclipse.che.api.workspace.server.wsplugins.model;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
 /** Represents Che plugin in sidecar-powered workspace. */
-public class ChePlugin extends PluginBase {
-  private List<EditorCompatibility> editors = new ArrayList<>();
+public class ChePlugin {
 
-  public ChePlugin editors(List<EditorCompatibility> editors) {
-    this.editors = editors;
+  private String name = null;
+  private String publisher = null;
+  private String id = null;
+  private String version = null;
+  private List<CheContainer> containers = new ArrayList<>();
+  private List<ChePluginEndpoint> endpoints = new ArrayList<>();
+
+  @JsonProperty("workspaceEnv")
+  private List<EnvVar> workspaceEnv = new ArrayList<>();
+
+  /** Object name. Name must be unique. */
+  public ChePlugin name(String name) {
+    this.name = name;
+    return this;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public ChePlugin id(String id) {
+    this.id = id;
+    return this;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public ChePlugin version(String version) {
+    this.version = version;
+    return this;
+  }
+
+  public String getPublisher() {
+    return publisher;
+  }
+
+  public void setPublisher(String publisher) {
+    this.publisher = publisher;
+  }
+
+  public ChePlugin publisher(String publisher) {
+    this.publisher = publisher;
     return this;
   }
 
-  public List<EditorCompatibility> getEditors() {
-    if (editors == null) {
-      editors = new ArrayList<>();
+  public String getVersion() {
```

Comment:
Method order could be a more consistent here, e.g have the chained method, then `set` then `get`.